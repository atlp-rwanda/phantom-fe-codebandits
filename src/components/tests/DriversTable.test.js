import DriversTable from '@components/DriversTable.js';
import { render } from '@testing-library/react';
import ProviderWrapper from '@utils/TestUtil.js';
import React from 'react';

describe('Driver Table', () => {
  const data = [
    {
      id: 2,
      name: 'Muhire Girbert',
      nationalid: '1199721772809249',
      mobileNumber: '0788251891',
      email: 'ctruswell1@histats.com',
      license: '1195510237288508',
      address: 'Kimihurura',
      assigned_bus: 'RAB1000',
      assigned_route: null
    },
    {
      email: 'driver@me.com',
      mobileNumber: '0788498955',
      company: 'roy',
      address: 'Nyamirambo',
      nationalid: '1199780030880040',
      license: '1199780030880040',
      name: 'Fabrice Hategekimana',
      assigned_bus: null,
      assigned_route: null,
      id: 5
    }
  ];
  it('Render the table', () => {
    const elem = render(
      <ProviderWrapper
        children={<DriversTable data={data} />}
      ></ProviderWrapper>
    );
    expect(elem).toMatchSnapshot();
  });
});
