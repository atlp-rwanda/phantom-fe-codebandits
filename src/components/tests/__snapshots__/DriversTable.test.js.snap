// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Driver Table Render the table 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex justify-between content-center pb-7"
      >
        <div>
          <input
            class="p-1 rounded-md border-2"
            name="search"
            placeholder="Search Drivers..."
            type="text"
            value=""
          />
        </div>
        
      </section>
      <table
        class="w-full overflow-x-scroll font-raleway"
        role="table"
      >
        <thead>
          <tr
            class="pb-10 "
            role="row"
          >
            <th
              class="text-left font-extrabold pr-5 border-b-2 pb-4"
              colspan="1"
              role="columnheader"
            >
              Names
            </th>
            <th
              class="text-left font-extrabold pr-5 border-b-2 pb-4"
              colspan="1"
              role="columnheader"
            >
              Email
            </th>
            <th
              class="text-left font-extrabold pr-5 border-b-2 pb-4"
              colspan="1"
              role="columnheader"
            >
              License
            </th>
            <th
              class="text-left font-extrabold pr-5 border-b-2 pb-4"
              colspan="1"
              role="columnheader"
            >
              Assigned Bus
            </th>
            <th
              class="text-left font-extrabold pr-5 border-b-2 pb-4"
              colspan="1"
              role="columnheader"
            >
              Management
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              Muhire Girbert
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              ctruswell1@histats.com
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              1195510237288508
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              <a
                class="text-primary"
                href="/#buses"
              >
                RAB1000
              </a>
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              <div
                class="z-10 font-sm"
              >
                <select
                  class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
                  id="select"
                  name=""
                  role="select"
                >
                  <option
                    hidden=""
                    value="manage"
                  >
                    Manage
                  </option>
                  <option
                    value="view"
                  >
                    View
                  </option>
                  
                  
                  
                  
                </select>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              Fabrice Hategekimana
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              driver@me.com
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              1199780030880040
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              None
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              <div
                class="z-10 font-sm"
              >
                <select
                  class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
                  id="select"
                  name=""
                  role="select"
                >
                  <option
                    hidden=""
                    value="manage"
                  >
                    Manage
                  </option>
                  <option
                    value="view"
                  >
                    View
                  </option>
                  
                  
                  
                  
                </select>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <section
      class="flex justify-between content-center pb-7"
    >
      <div>
        <input
          class="p-1 rounded-md border-2"
          name="search"
          placeholder="Search Drivers..."
          type="text"
          value=""
        />
      </div>
      
    </section>
    <table
      class="w-full overflow-x-scroll font-raleway"
      role="table"
    >
      <thead>
        <tr
          class="pb-10 "
          role="row"
        >
          <th
            class="text-left font-extrabold pr-5 border-b-2 pb-4"
            colspan="1"
            role="columnheader"
          >
            Names
          </th>
          <th
            class="text-left font-extrabold pr-5 border-b-2 pb-4"
            colspan="1"
            role="columnheader"
          >
            Email
          </th>
          <th
            class="text-left font-extrabold pr-5 border-b-2 pb-4"
            colspan="1"
            role="columnheader"
          >
            License
          </th>
          <th
            class="text-left font-extrabold pr-5 border-b-2 pb-4"
            colspan="1"
            role="columnheader"
          >
            Assigned Bus
          </th>
          <th
            class="text-left font-extrabold pr-5 border-b-2 pb-4"
            colspan="1"
            role="columnheader"
          >
            Management
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            Muhire Girbert
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            ctruswell1@histats.com
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            1195510237288508
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            <a
              class="text-primary"
              href="/#buses"
            >
              RAB1000
            </a>
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            <div
              class="z-10 font-sm"
            >
              <select
                class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
                id="select"
                name=""
                role="select"
              >
                <option
                  hidden=""
                  value="manage"
                >
                  Manage
                </option>
                <option
                  value="view"
                >
                  View
                </option>
                
                
                
                
              </select>
            </div>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            Fabrice Hategekimana
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            driver@me.com
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            1199780030880040
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            None
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            <div
              class="z-10 font-sm"
            >
              <select
                class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
                id="select"
                name=""
                role="select"
              >
                <option
                  hidden=""
                  value="manage"
                >
                  Manage
                </option>
                <option
                  value="view"
                >
                  View
                </option>
                
                
                
                
              </select>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
